indextools/pds4_index_tools.py

```
@@ -0,0 +1,361 @@
+    variable. If 'LID' is in labels, the 'LID' key will be given lid as a value. If 'Path'
+    is in labels, the 'Path' key will be given filepath as a value. All other labels will
+    be given None as values until they are replaced with their actual values in
+    add_bundle_data (for bundle products) or add_collection_data (for collection
+    products). This also creates the final version of member_index in the case of
+    bundlesets.
+    
+    
+
+    Inputs:
+        labels          The labels to put into member_index.
+
+        filepath        The path to the data product. If none exists, just
+                        use None
+
+        lid             The LID of the indexed data. This will be used as the identifier
+                        for crossmatching.
+        
+        member_index    The dictionary that will contain the information of the data
+                        products.
+    """
+    temp_dict = {}
+    lid_short = lid.split(':')[-1]
+    for label in labels:
+        if label == 'LID':
+            temp_dict.update({label: lid})
+        elif label == 'Path':
+            temp_dict.update({label: filepath})
+        else:
+            temp_dict.update({label: None})
+    member_index[lid_short] = temp_dict
+
+
+def create_results_file(base_directory, keyword, member_index):
+    """Create the file of the results.
+    
+    The file is opened within the base_directory and contains the keyword in the name.
+    The labels are taken from the keys of the nested dictionaries within member_index,
+    and are used as fieldnames for the csv file. Each nested dictionary within
+    member_index is then entered as a row into the csv file.
+
+    Inputs:
+        base_directory    The path to the base directory.
+
+        keyword           The keyword to determine the name of the file.
+
+        member_index      The index of data product information.
+    """
+    found_keys = list(member_index.keys())[0]
```

What happens if `member_index` is empty?

---

I have put in an extra warning to check if member_index does not populate.

---

This is another case of a library function exiting the program. I don't think this should ever happen. It should raise an exception and let the caller deal with it. Perhaps the caller wants to do something other than exit...  You could just convert this to a ValueError exception using the same error message.